File: Lecture17/higgs.cpp
#include <cmath>
#include <cstdlib>
#include <iostream>
using namespace std;

#include "nonlin.hpp"
using namespace cpt;

double V(double x)
{
    return -pow(x, 2.0)/ 2 + pow(x, 4.0)/4;
}

int main()
{
    double acc = 1e-6;
    double guess1 = 0.1, guess2 = -0.1;
    double x_min;
    double V_min = find_minimum< double (*)(double)> (guess1, guess2, V, acc, x_min);
    cout << "V(x) = " << V_min << " at x = " << x_min << endl;

    double x_max;
    double V_max = find_maximum< double (*)(double)> (guess1, guess2, V, acc, x_max);
    cout << "V(x) = " << V_max << " at x = " << x_max << endl;
}

File: Lecture17/higgs.py
from cpt import *

def V(x) : 
    return -pow(x, 2.0)/ 2 + pow(x, 4.0)/4



acc = 1e-6
guess1 = 0.1
guess2 = -0.1
[x_min, V_min] = find_minimum(guess1, guess2, V, acc)
print "V(x) = " + str(V_min)+  " at x = " + str(x_min)

[x_max, V_max] = find_maximum (guess1, guess2, V, acc)
print "V(x) = " + str(V_max)+  " at x = " + str(x_max)
